lib/src/simulations/rose_pattern.dart

```diff
@@ -274,8 +269,7 @@ class _RoseState extends State<Rose> {
 
     setState(() {
       if(!(widget.d==0)&&!(widget.d==0&&widget.n==0)){
```

Comment:
```suggestion
      if(loopi == 0) {
        looplength = 2 * pi * widget.d;
      }
      if(!(widget.d==0)&&!(widget.d==0&&widget.n==0)){
```

`looplength` depends on `widget.d` that's why the first animation is always short, because initially it's value is 2pi and is not multiplied by `widget.d`.